tools/codegen/pkg/schemacheck/generator.go

```diff
@@ -55,10 +64,16 @@ func (g *generator) GenGroup(groupCtx generation.APIGroupContext) error {
 
 	errs := []error{}
 
+	comparatorOptions := options.NewComparatorOptions()
```

Comment:
wow, that integrated deeply.  You sure you want to hook in this deep?

Reply:
We are hooking in "this deep" for other generators so that we can supply the inputs needed. The codegen tool goes through all the paths and extracts the files needed, and can then pull into memory the other inputs, in this case, the previous file version.

I did try with this one to create temp files but in the end, this was actually easier to achieve this way, plus, we get any errors back as an `err` rather than whatever stderr output the tool would give, so we have more control over how we report that back to the end user.

Given we committed to going this deep with others, I don't see it as an issue here. I've made my bed as it were 😅